doc/FLINT models.md

```diff
@@ -0,0 +1,404 @@
+2. `acts` - An array of **actions** that may be taken during evaluation of the process
+3. `facts` - An array of **facts** that can be referenced within the model
+4. `duty` - An array of **duties** that can describe agreements between actors and enforce certain actions when this duty is broken
+
+### Model
+
+The `model` property contains a description that describes the processes that is modeled by the FLINT model.
+
+
+```json
+{
+    "model": "Fictieve kinderbijslag"
+}
+```
+### Acts (actions)
+
+The `acts` property contains an array of actions that may be **taken** during the process. An act is defined by an object that describes among others: the actor that will take the action; the recipient that receives the result; the conditions that must be met in order to take the act.
+
+> On the implementation level the `recipient` expresses a need (with help of the [discipl-abundance-service](https://github.com/discipl/discipl-abundance-service/)) and the `actor` attends to this need in order to fulfill it. So the actors in the model are referring to entities in the abundance service.
+
+An act consists out of the following properties:
+
+- `act` - A unique sentence that describes the act and is used to reference the action within the model. This sentence is usually enclosed between `<<` and `>>` to indicate that the sentence is referring to an act.
+- `actions` - One or few words denoting what action is done on the object.
+- `actor` - A reference to a fact that defines the actor that can take this action.
+- `recipient` - A reference to a fact that defines the recipient of the result of this action.
+- `object` - The object on which the action is executed by the actor, and that is needed by the recipient of this action. The object can be a entity or agent or a reference to such via a fact.  The object can also be a reference to any other fact or even an action and duty referring to their corresponding state.
+- `preconditions` - Boolean logic with references to other facts defined in the model. The evaluation of this expression determents if the actions can be taken or not.
+- `create` - An array of references to other acts, facts and/or duties that will follow out of this action. The creation of a fact out of a action can be enforced by defining that fact with a `CREATE` expression as function.
+- `terminate` - An of references to other actions, facts and/or duties that will be terminated (eg. revoked) by taking this action.
+- `reference` -  A jurisdictional reference to law/regulation this action is implied from.
+
+```json
+{
+    "act": "<<kinderbijslag aanvragen>>",
+    "actor": "[ouder]",
+    "action": "[aanvraag indienen]",
+    "object": "[verzoek]",
+    "recipient": "[minister]",
+    "preconditions": {
+        "expression": "AND",
+        "operands": [
+            "[ouder heeft kinderen]"
+        ]
+    },
+    "create": [
+        "[aanvraag]"
+    ],
+    "terminate": [],
+    "reference": "rt. 1:2 lid 1"
```

Comment:
This has been changed to a "sources" array. This has no implications for execution context, so I hadn't changed it for some of the tests.

https://github.com/discipl/discipl-law-reg/blob/tests/rvo-tegemoetkoming/test/tegemoetkoming-schade-covid19.flint.json is the most recent model which has examples of this.

Reply:
I replaced the `reference` properties with the `source` array as described in the covid19 flint example